src/main/java/randoop/contract/ObserverEqArray.java

```diff
@@ -0,0 +1,211 @@
+      return true;
+    }
+    return PrimitiveTypes.isBoxedPrimitive(cls) || cls.isPrimitive();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (o == this) {
+      return true;
+    }
+    if (!(o instanceof ObserverEqArray)) {
+      return false;
+    }
+    ObserverEqArray other = (ObserverEqArray) o;
+    return Arrays.equals(toObjectArray(value), toObjectArray(other.value));
+  }
+
+  @Override
+  public int hashCode() {
+    return Arrays.hashCode(toObjectArray(value));
+  }
+
+  @Override
+  public String toCodeString() {
+    if (value.getClass().getComponentType() == float.class
+        || value.getClass().getComponentType() == double.class) {
+      return String.format(
+          "org.junit.Assert.assertArrayEquals(x0, %s, %s);",
+          newArrayExpression(), FLOATING_POINT_DELTA);
+    } else {
+      return String.format("org.junit.Assert.assertArrayEquals(x0, %s);", newArrayExpression());
+    }
+  }
+
+  /**
+   * Returns a Java array constructor expression for this, e.g., "new int[] {1,2,3}".
+   *
+   * @return a Java array constructor expression for this
+   */
+  private String newArrayExpression() {
+    return "new " + value.getClass().getCanonicalName() + arrayComponentsToString();
+  }
+
+  /**
+   * Converts an Object into its array representation.
+   *
+   * @param value the given Object
+   * @return its array representation
+   */
+  private Object[] toObjectArray(Object value) {
+    if (value instanceof byte[]) {
+      return ArrayUtils.toObject((byte[]) value);
+    } else if (value instanceof short[]) {
+      return ArrayUtils.toObject((short[]) value);
+    } else if (value instanceof int[]) {
+      return ArrayUtils.toObject((int[]) value);
+    } else if (value instanceof long[]) {
+      return ArrayUtils.toObject((long[]) value);
+    } else if (value instanceof float[]) {
+      return ArrayUtils.toObject((float[]) value);
+    } else if (value instanceof double[]) {
+      return ArrayUtils.toObject((double[]) value);
+    } else if (value instanceof char[]) {
+      return ArrayUtils.toObject((char[]) value);
+    } else if (value instanceof boolean[]) {
+      return ArrayUtils.toObject((boolean[]) value);
+    } else {
+      return (Object[]) value;
+    }
+  }
+
+  /**
+   * Returns a string representation of the components of the array, enclosed in curly braces, as it
+   * would appear in Java source code.
+   *
+   * @return a String that represents the components of the array
+   */
+  private String arrayComponentsToString() {
+    StringJoiner sj = new StringJoiner(", ", "{", "}");
+    int length = Array.getLength(value);
+    for (int i = 0; i < length; i++) {
+      sj.add(literalValueToString(Array.get(value, i)));
+    }
+    return sj.toString();
+  }
+
+  /**
+   * Returns one literal value, as it would appear in Java code
+   *
+   * @return the Java code for the literal value
+   */
+  private String literalValueToString(Object element) {
```

Comment:
I don't think this new method is necessary.  It's better to extend `Value.toCodeString()` and then use it in `arrayComponentsToString()`.

Reply:
When I was changing Value's toCodeString() method to include Classes and Enums, I noticed that there was a method in the Value class called arrayLengthOk(Object a) that "returns true if the given array is shorter than the --string-maxlen=N parameter." Should I use this method instead of the constant MAX_ARRAY_LENGTH in RegressionCaptureGenerator?